resources/qml/WelcomePages/AddPrinterBySelectionContent.qml

```diff
@@ -0,0 +1,154 @@
+        anchors.horizontalCenter: parent.horizontalCenter
+        horizontalAlignment: Text.AlignHCenter
+        text: catalog.i18nc("@label", "Add a printer")
+        color: UM.Theme.getColor("primary_button")
+        font: UM.Theme.getFont("large_bold")
+        renderType: Text.NativeRendering
+    }
+
+    DropDownWidget
+    {
+        id: addNetworkPrinterDropDown
+
+        anchors.top: titleLabel.bottom
+        anchors.left: parent.left
+        anchors.right: parent.right
+        anchors.margins: 20
+
+        title: catalog.i18nc("@label", "Add a networked printer")
+        contentShown: true  // by default expand the network printer list
+
+        onClicked:
+        {
+            if (contentShown)
+            {
+                addLocalPrinterDropDown.contentShown = false
+            }
+        }
+
+        contentComponent: networkPrinterListComponent
+
+        Component
+        {
+            id: networkPrinterListComponent
+
+            AddNetworkPrinterScrollView
+            {
+                id: networkPrinterScrollView
+
+                maxItemCountAtOnce: 6  // show at max 6 items at once, otherwise you need to scroll.
+
+                onRefreshButtonClicked:
+                {
+                    UM.OutputDeviceManager.startDiscovery()
+                }
+
+                onAddByIpButtonClicked:
+                {
+                    base.gotoPage("add_printer_by_ip")
+                }
+            }
+        }
+    }
+
+    DropDownWidget
+    {
+        id: addLocalPrinterDropDown
+
+        anchors.top: addNetworkPrinterDropDown.bottom
+        anchors.left: parent.left
+        anchors.right: parent.right
+        anchors.margins: 20
+
+        title: catalog.i18nc("@label", "Add a non-networked printer")
+
+        onClicked:
+        {
+            if (contentShown)
+            {
+                addNetworkPrinterDropDown.contentShown = false
+            }
+        }
+
+        contentComponent: localPrinterListComponent
+
+        Component
+        {
+            id: localPrinterListComponent
+
+            AddLocalPrinterScrollView
+            {
+                id: localPrinterView
+
+                maxItemCountAtOnce: 10  // show at max 10 items at once, otherwise you need to scroll.
+            }
+        }
+    }
+
+    Cura.PrimaryButton
+    {
+        id: nextButton
+        anchors.right: parent.right
+        anchors.bottom: parent.bottom
+        anchors.margins: 40
+        enabled:
+        {
+            // If the network printer dropdown is expanded, make sure that there is a selected item
+            if (addNetworkPrinterDropDown.contentShown)
+            {
+                return addNetworkPrinterDropDown.contentItem.currentItem != null
+            }
+            else
+            {
+                return addLocalPrinterDropDown.contentItem.currentItem != null
+            }
+        }
+
+        text: catalog.i18nc("@button", "Next")
+        width: 140
```

Comment:
Isn't there a default size for primary buttons?

Reply:
Not for the width.

Reply:
This is fixed in https://github.com/Ultimaker/Cura/pull/5470